src/js/services/blob.service.js

```diff
@@ -0,0 +1,348 @@
+
+    if (Array.isArray(context)) {
+      if (part === '-') {
+        part = context.length;
+      } else if (part % 1 !== 0 && part >= 0) {
+        throw new Error('Invalid pointer, array element segments must be a positive integer, zero or \'-\'');
+      }
+    } else if (typeof context !== 'object') {
+      return null;
+    } else if (!context.hasOwnProperty(part)) {
+      // Some opcodes create the path as they're going along
+      if (op === 'set') {
+        context[part] = {};
+      } else if (op === 'unshift') {
+        context[part] = [];
+      } else {
+        return null;
+      }
+    }
+
+    if (pointer.length !== 0) {
+      return this._traverse(context[part], pointer, originalPointer, op, params);
+    }
+
+    switch (op) {
+      case 'set':
+        context[part] = params[0];
+        break;
+      case 'unset':
+        if (Array.isArray(context)) {
+          context.splice(part, 1);
+        } else {
+          delete context[part];
+        }
+        break;
+      case 'extend':
+        if (typeof context[part] !== 'object') {
+          throw new Error('Tried to extend a non-object');
+        }
+        // extend(true, context[part], params[0]);
+        $.extend(context[part], params[0]);
+        break;
+      case 'unshift':
+        if (typeof context[part] === 'undefined') {
+          context[part] = [];
+        } else if (!Array.isArray(context[part])) {
+          throw new Error('Operator "unshift" must be applied to an array.');
+        }
+        context[part].unshift(params[0]);
+        break;
+      case 'filter':
+        if (Array.isArray(context[part])) {
+          context[part].forEach(function(element, i) {
+            if (typeof element === 'object' && element.hasOwnProperty(params[0]) && element[params[0]] === params[1]) {
+              var subpointer = originalPointer + '/' + i;
+              var subcommands = normalizeSubcommands(params.slice(2));
+
+              subcommands.forEach(function(subcommand) {
+                var op = subcommand[0];
+                var pointer = subpointer + subcommand[1];
+                _this.applyUpdate(op, pointer, subcommand.slice(2));
+              });
+            }
+          });
+        }
+        break;
+      default:
+        throw new Error('Unsupported op ' + op);
+    }
+  };
+
+  BlobObj.prototype.escapeToken = function(token) {
+    return token.replace(/[~\/]/g, function(key) {
+      return key === '~' ? '~0' : '~1';
+    });
+  };
+
+  BlobObj.prototype.unescapeToken = function(str) {
+    return str.replace(/~./g, function(m) {
+      switch (m) {
+        case '~0':
+          return '~';
+        case '~1':
+          return '/';
+      }
+      throw new Error('Invalid tilde escape: ' + m);
+    });
+  };
+
+  /**
+   * Sumbit update to blob vault
+   */
+
+  BlobObj.prototype.postUpdate = function(fn) {
+
+    // Callback is optional
+    if (typeof fn !== 'function') {
```

Comment:
Shorter version would be 
`if ("function" !== typeof callback) callback = $.noop;`